### PR TITLE
Improve Reverb includes, remove Qt dependency

### PR DESCRIPTION
**Pull request by daschuer**
This does not fix the original issue, but improves compile time a bit.

### CODE REVIEW

File: lib/reverb/basics.h [hunk 1]
```diff
@@ -29,78 +31,16 @@
...
-
 #include <cmath>
-
-#include <stdlib.h>
-#include <string.h>
-
-#include <math.h>
-#include <float.h>
-
+#include <cstring>
```
Comment (Swiftb0y) [inline lib/reverb/basics.h:35]: seems unnecessary

File: lib/reverb/basics.h [hunk 2]
```diff
@@ -29,78 +31,16 @@
 	02111-1307, USA or point your web browser to http://www.gnu.org.
 */
 
-#ifndef BASICS_H
-#define BASICS_H
-
 #include <cmath>
```
Comment (Swiftb0y) [inline lib/reverb/basics.h:34]: can also be removed I think, right?

**Comment by daschuer** [comment]
I have added comments to clarify the reason for the include.

File: lib/reverb/basics.h (hunk 2 above)
Comment (daschuer) [reply lib/reverb/basics.h]: No, we need it for M_PI

File: lib/reverb/basics.h (hunk 1 above)
Comment (daschuer) [reply lib/reverb/basics.h]: required for memset()

**Comment by daschuer** [comment]
Rebased to make clazy work.

**Review by Swiftb0y — COMMENTED** [review]
Thank you, but the questions remain.

File: lib/reverb/basics.h [hunk 3]
```diff
@@ -29,78 +31,16 @@
...
-typedef qint64 int64;
-typedef quint64 uint64;
-
-#define MIN_GAIN 1e-6 /* -120 dB */
-/* smallest non-denormal 32 bit IEEE float is 1.18e-38 */
-#define NOISE_FLOOR 1e-20 /* -400 dB */
-
-/* //////////////////////////////////////////////////////////////////////// */
+#include <cmath> // for M_PI
+#include <cstring> // for memset
```
Comment (Swiftb0y) [inline lib/reverb/basics.h:35]: Why is this in the header if neither `M_PI` nor `memset` are used in it?

Comment (daschuer) [reply lib/reverb/basics.h]: I have moved them closer to usage.

**Comment by daschuer** [comment]
Done.

**Review by Swiftb0y — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**